contracts/ServiceNodeRewards.sol

```diff
@@ -755,11 +797,40 @@ contract ServiceNodeRewards is Initializable, Ownable2StepUpgradeable, PausableU
     /// the owner.
     /// @param newMax The new maximum non-signer threshold
     function setBLSNonSignerThresholdMax(uint256 newMax) public onlyOwner {
-        require(newMax > 0, "The new BLS non-signer threshold must be non-zero");
+        if (newMax <= 0)
+            revert PositiveNumberRequirement();
         blsNonSignerThresholdMax = newMax;
         emit BLSNonSignerThresholdMaxUpdated(newMax);
     }
 
+    /// @notice Max Claim amount to use in the Exponential Moving Average check
+    /// before allowing the user to claim. If the claimed amount over 24 hours
+    /// exceeds this then the claim will fail
+    /// @param newMax The new maximum non-signer threshold
+    function setClaimEMAThreshold(uint256 newMax) public onlyOwner {
+        if (newMax <= 0)
+            revert PositiveNumberRequirement();
+        claimEMAThreshold = newMax;
+        emit ClaimEMAThresholdUpdated(newMax);
+    }
+
+    /// @notice Updates the Exponential Moving Average (EMA) with a new value
+    /// @dev This function calculates a time-weighted EMA using the provided new value and current time
+    /// @param ema The TimeWeightedEMA struct to update
+    /// @param newValue The new value to incorporate into the EMA.
+    /// @param currentTime The current timestamp, usually `block.timestamp`
+    /// @custom:formula EMA = EMA_old + α * (newValue - EMA_old)
```

Comment:
This formula is out of date -> updated to not adjust the new value

Reply:
There might be a misunderstanding of what the EMA is doing here. 

The rate limit is triggered if the EMA goes over the fixed limit of 2M.

We cant replace the EMA with a fixed limit, the EMA is addressing the issue of having to track the history of previous claims.

Reply:
Oh sorry yeah, the EMA is applied on the claims amount over 24 hr period not the claim limit, that makes it less of a problem. Kee's suggestion though was just to hard reset every 24 hours to keep it simple.

Reply:
I think this is somewhat ok, I prefer the simpler hard-reset for predictability. If it's been cleared with Kee then I have no problems.